### PR TITLE
MG-2136 - Generate mocks with mockery for Bootstrap service 

### LINKED ISSUE #1993: Feature: Generate Mocks Per Service
### Is your feature request related to a problem? Please describe.

The problem we are currently having is that we write mocks for interfaces ourselves which might be cumbersome and had to maintain especially for SDK.

### Describe the feature you are requesting, as well as the possible use case(s) for it.

Instead of writing down each mock for individual interfaces use https://github.com/vektra/mockery or https://github.com/uber-go/mock to generate mocks for interfaces. This will be instrumental in testing. This is quite useful for services depending on the SDK for example, bootstrap and certs service. This approach simplifies the testing process as initially, one would need to generate the things services, then the things server and finally use the things server URL on the SDK. This effective means that when testing other services you inherently test services that depend on the SDK which is not ideal. With SDK mocks you can mock the SDK rather than mocking dependent services.

### Indicate the importance of this feature to you.

Nice-to-have

### Anything else?

Initial work is done at https://github.com/absmach/magistrala-old/pull/127

## Procedure

- [x] auth
- [x] bootstrap https://github.com/absmach/magistrala/issues/2136
- [x] http - https://github.com/absmach/magistrala/issues/2170
- [x] invitations - https://github.com/absmach/magistrala/issues/2143
- [x] mqtt - https://github.com/absmach/magistrala/issues/2161
- [x] pkg
- [x] provision
- [x] things
- [x] users - https://github.com/absmach/magistrala/issues/2164

- [x] certs - https://github.com/absmach/magistrala/issues/2137
- [x] consumers - https://github.com/absmach/magistrala/issues/2147
- [x] lora - https://github.com/absmach/magistrala/issues/2145
- [x] readers - https://github.com/absmach/magistrala/issues/2166
- [x] twins - https://github.com/absmach/magistrala/issues/2173

Service/modules which have no mocks before
- ws
- opcua
- coap
- logger
- internal

### PR DESCRIPTION
**Pull request by nyagamunene**
# What type of PR is this?
This is a feature because its add the following functionality: generates mocks for bootstrap.

## What does this do?
It generates the mocks for bootstrap service.

## Which issue(s) does this PR fix/relate to?

- Related to #2136

## Have you included tests for your changes?

Yes, I have included tests.

## Did you document any new/modified feature?

No, I have not updated the documentation.

### CODE REVIEW

**Comment by WashingtonKK** [comment]
Please sign off your commits, check DCO report [here](https://github.com/absmach/magistrala/pull/2128/checks?check_run_id=23168073503) and fix.

**Review by WashingtonKK — CHANGES_REQUESTED** [review]
We do not need mocks for the ws and coap services since they are not used. Only replace manually written mocks with generated mocks.
An example is bootstrap service, we have manually written mocks for the service but we need to have them generated instead.

**Comment by WashingtonKK** [comment]
Also update the mockery version used to use [the latest version](https://github.com/vektra/mockery/releases/tag/v2.42.1)

File: bootstrap/configs.go [hunk 1]
```diff
@@ -60,6 +60,8 @@ type ConfigsPage struct {
 }
 
 // ConfigRepository specifies a Config persistence API.
+//
+//go:generate mockery --name ConfigRepository --output=./mocks --filename configs.go --quiet --note "Copyright (c) Abstract Machines"
```
Comment (rodneyosodo) [inline bootstrap/configs.go:64]: Run 
```bash
make mocks
```

to generate the mocks

**Review by rodneyosodo — CHANGES_REQUESTED** [review]

File: bootstrap/configs.go (hunk 1 above)
Comment (nyagamunene) [reply bootstrap/configs.go:64]: Okay

**Pull request closed without merging**